server/routes/ad.ts

```diff
@@ -461,6 +460,16 @@ const getDetectors = async (
           ? DETECTOR_STATE.UNEXPECTED_FAILURE
           : DETECTOR_STATE.INIT_FAILURE;
       }
+
+      /*
+        If a detector has no features, set to a feature required state
+      */
+      if (
+        detectorState.state === DETECTOR_STATE.DISABLED &&
+        finalDetectors[i].featureAttributes.length === 0
+      ) {
+        detectorState.state = DETECTOR_STATE.FEATURE_REQUIRED;
+      }
     });
```

Comment:
you may encapsulate such logic to a method with appropriate name like `getStateFromRawResponse()` or better name

Reply:
Sure, good idea. Will add a function in adHelpers.